### PR TITLE
Python rewrite

### PR DESCRIPTION
**Pull request by htgoebel**
Hi, here is the Python-rewrite of the debops-scripts.

Notable changed related to the shell-version:
- Playbooks are no longer searched in `/usr/local/share/debops` nor in
  `/usr/share/debops`. We assume DebOps is used from a user account,
  so installing playbooks globally is not the common case.
- Sourcing `.debops.cfg` is no longer supported, `.debops.cfg` now is
  assumed to be an ini-file.
- The `ansible_config_hook` (which was undocumented anyway) is gone.
  Instead you can put sections `[ansible ...]` into `.debops.cfg`
  which will go into `ansible.cfg`. This allows for easy adding e.g. a
  `[paramiko]` section to `ansible.cfg`.
- The padlock-script is no longer used to decide if secrets are
  encrypted. This is now done by testing for the encrypted keyfile and
  the encrypted encfs-config.
- The padlock script still exists and has been simplified a lot. Most
  functionality has been moved into the DebOps python library.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # Part of the DebOps project - http://debops.org/
 
 
-PREFIX?="/usr/local"
+PREFIX="/usr/local"
```
Comment (drybjed) [inline Makefile]: IIRC, `PREFIX ?=` allows you to se the prefix directory in the command line, so that running

```
PREFIX=/opt make
```

becomes possible. Is that change needed here?

Comment (drybjed) [inline Makefile]: OK, I've remembered your previous PR, nevermind. :)

**Comment by drybjed** [comment]
@htgoebel Thanks for an awesome rewrite! :-) I want to check it out in action before merging, however I'm not sure how the install procedure has changed...

```
drybjed@test ~/new/debops/ python-rewrite!? % sudo ./setup.py install
running install
Checking .pth file support in /usr/local/lib/python2.7/dist-packages/
/usr/bin/python -E -c pass
TEST PASSED: /usr/local/lib/python2.7/dist-packages/ appears to support .pth files
running bdist_egg
running egg_info
writing requirements to debops.egg-info/requires.txt
writing debops.egg-info/PKG-INFO
writing top-level names to debops.egg-info/top_level.txt
writing dependency_links to debops.egg-info/dependency_links.txt
error: package directory 'debops' does not exist
```

How can I try it out?

File: bin/debops-padlock [hunk 2]
```diff
@@ -26,227 +26,122 @@
...
-require_commands () {
-  for name in ${@} ; do
-    if ! type ${name} > /dev/null 2>&1 ; then
-      error_msg "${name}: command not found"
-    fi
-  done
-}
+devrandom = os.environ.get('DEVRANDOM', "/dev/urandom")
 
+SCRIPT_FILENAME = os.path.join(os.path.dirname(__file__), 'padlock.py')
```
Comment (drybjed) [inline bin/debops-padlock]: If I'm not mistaken, when scripts are installed in `/usr/local/bin/`, script would look there for `padlock.py` right? Could that script be included in a heredoc instead and outputted in place inside EncFS encrypted directory?

**Comment by htgoebel** [comment]
@drybjed The setup.py was unfinished. It should work now.

And I expect you are testing it in depth. I've already done a lot, but I would be astound if this is really free of errors.

File: bin/debops-padlock (hunk 2 above)
Comment (htgoebel) [inline bin/debops-padlock]: This is now changed, too.
heredocs are for Shells, Pythons like real food.

**Comment by drybjed** [comment]
@htgoebel OK, debops installed, now I'm running `debops-update`, but it looks like if playbooks haven't been installed anywhere, script just exits. Could you set a default install path to `~/.local/share/debops` if debops playbooks haven't been installed? I suppose that could be also done via Makefile target.

Also, `chmod +x setup.py` seems to be in order :-)

**Comment by drybjed** [comment]
@htgoebel could you rebase `debops-rewrite` branch on current `debops/debops:master`? There were some important fixes, like `--recursive` added in `bootstrap-ansible.sh`.

**Comment by drybjed** [comment]
@htgoebel Looks like `debops` script correctly finds the playbook, but incorrectly executes Ansible with it:

```
% debops common -t auth -vvv                                                              
Running Ansible playbook from:
/home/drybjed/.local/share/debops/debops-playbooks/playbooks/common.yml ...
ERROR: the playbook: common could not be found
```

```
% debops -t auth -vvv 
Running Ansible playbook from:
/home/drybjed/.local/share/debops/debops-playbooks/playbooks/site.yml ...

PLAY [Common configuration for all hosts] ************************************* 

GATHERING FACTS *************************************************************** 
```

**Comment by htgoebel** [comment]
@drybjed Thanks for reporting back
https://github.com/debops/debops/pull/67#issuecomment-66893540: both done
https://github.com/debops/debops/pull/67#issuecomment-66893739: rebase is not necessary, I have not changed anything in this area. We are using git, not old svn, so merging will not overwrite other changes :-)
https://github.com/debops/debops/pull/67#issuecomment-66893944: should be fixed now, too.

**Comment by drybjed** [comment]
@htgoebel It looks like everything works now, awesome job! I haven't tested `debops-padlock` yet and support for secrets.

Do you want to update the Makefile for new install procedure before I merge this, or should we do it later?

**Comment by htgoebel** [comment]
The makefile is an issue, more concrete: the PREFIX. We should discuss this on IRC.
